Libraries/Network/XMLHttpRequestBase.js

```diff
@@ -296,6 +319,20 @@ class XMLHttpRequestBase {
       onload(null);
     }
   }
+
+  _sendOnTimeout(): void {
```

Comment:
nit, you can consolidate these to something like `_sendEvent(this.ontimeout)` so you get rid of the duplication. then just call these from `setReadyState()`